src/Evolution/Systems/ScalarAdvection/VelocityField.cpp

```diff
@@ -5,7 +5,7 @@
 
 #include <cstddef>
 
-#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
 #include "DataStructures/Tensor/TypeAliases.hpp"
```

Comment:
you don't need `TypeAliases.hpp` in the `cpp` file, just the `hpp`, but you do need `DataVector.hpp` in the `cpp` file

The reason is that in the cpp you're actually using the class `DataVector` and so you need a full definition, which is provided in `DataVector.hpp`. `Tensor/TypeAliases.hpp` just does a bunch of forward declarations, while `Tensor.hpp` contains the definitions, so you need `Tensor.hpp` included in the `cpp`. However, `Tensor.hpp` includes `TypeAliases.hpp` and so you don't need to separately include `TypeAliases.hpp`

Reply:
Ah, okay! Thank you!